File: backend/app/schemas/recovery.py
from pydantic import BaseModel, Field


class RecoveryPredictionResponse(BaseModel):
    transaction_id: int
    recovery_probability: float = Field(
        ge=0.0,
        le=1.0,
    )
    recoverable: bool
    threshold: float = Field(
        ge=0.0,
        le=1.0,
    )
    recommended_action: str
    priority: str
    reason: str


class RecoveryCaseResponse(BaseModel):
    recovery_case_id: int
    transaction_id: int
    recovery_probability: float = Field(
        ge=0.0,
        le=1.0,
    )
    recoverable: bool
    recommended_action: str
    priority: str
    reason: str
    status: str
    estimated_revenue: float


class RecoveryExecutionResponse(BaseModel):
    recovery_case_id: int
    action: str
    status: str
    result: str

class RecoveryCaseListResponse(BaseModel):
    recovery_case_id: int
    transaction_id: int
    recovery_probability: float = Field(
        ge=0.0,
        le=1.0,
    )
    reason: str
    status: str
    estimated_revenue: float

class RecoveryActionResponse(BaseModel):
    action_type: str
    status: str
    reason: str
    result: str | None


class RecoveryCaseDetailResponse(BaseModel):
    recovery_case_id: int
    transaction_id: int
    recovery_probability: float = Field(
        ge=0.0,
        le=1.0,
    )
    reason: str
    status: str
    estimated_revenue: float
    action: RecoveryActionResponse

class RecoveryActionHistoryResponse(BaseModel):
    action_id: int
    action_type: str
    status: str
    reason: str
    result: str | None
    executed_at: str | None


class TransactionItemResponse(BaseModel):
    id: int
    merchant_id: int
    customer_id: int
    customer_name: str | None = None
    customer_email: str | None = None
    amount: float
    currency: str
    status: str
    failure_reason: str | None
    payment_method: str
    retry_count: int
    created_at: str
    has_recovery_case: bool = False
    recovery_case_id: int | None = None
    recovery_case_status: str | None = None
    recovery_probability: float | None = None


class MetricsSummaryResponse(BaseModel):
    total_revenue_at_risk: float
    total_recoverable_revenue: float
    total_recovered_revenue: float
    recovery_rate_percent: float
    total_failed_transactions: int
    total_successful_transactions: int
    total_recovery_cases: int
    executed_recovery_cases: int
    failure_reasons_breakdown: dict[str, int]
    payment_methods_breakdown: dict[str, int]


class AuditEventResponse(BaseModel):
    id: int
    merchant_id: int
    recovery_case_id: int | None
    event_type: str
    actor: str
    description: str
    created_at: str

File: frontend/src/types/recovery.ts
export interface TransactionItem {
  id: number;
  merchant_id: number;
  customer_id: number;
  customer_name?: string | null;
  customer_email?: string | null;
  amount: number;
  currency: string;
  status: 'success' | 'failed' | string;
  failure_reason?: string | null;
  payment_method: 'upi' | 'card' | 'netbanking' | 'wallet' | string;
  retry_count: number;
  created_at: string;
  has_recovery_case: boolean;
  recovery_case_id?: number | null;
  recovery_case_status?: string | null;
  recovery_probability?: number | null;
}

export interface RecoveryPrediction {
  transaction_id: number;
  recovery_probability: number;
  recoverable: boolean;
  threshold: number;
  recommended_action: string;
  priority: 'high' | 'medium' | 'low' | string;
  reason: string;
}

export interface RecoveryCaseItem {
  recovery_case_id: number;
  transaction_id: number;
  recovery_probability: number;
  reason: string;
  status: 'open' | 'executed' | 'failed' | string;
  estimated_revenue: number;
  recommended_action?: string;
  priority?: string;
  recoverable?: boolean;
}

export interface RecoveryCaseDetail extends RecoveryCaseItem {
  action: {
    action_type: string;
    status: string;
    reason: string;
    result?: string | null;
  };
}

export interface RecoveryActionHistory {
  action_id: number;
  action_type: string;
  status: string;
  reason: string;
  result?: string | null;
  executed_at?: string | null;
}

export interface RecoveryExecutionResult {
  recovery_case_id: number;
  action: string;
  status: string;
  result: string;
}

export interface AgentAnalysisResult {
  transaction_id: number;
  analysis: string;
}

export interface DashboardMetrics {
  total_revenue_at_risk: number;
  total_recoverable_revenue: number;
  total_recovered_revenue: number;
  recovery_rate_percent: number;
  total_failed_transactions: number;
  total_successful_transactions: number;
  total_recovery_cases: number;
  executed_recovery_cases: number;
  failure_reasons_breakdown: Record<string, number>;
  payment_methods_breakdown: Record<string, number>;
}

export interface AuditEventItem {
  id: number;
  merchant_id: number;
  recovery_case_id?: number | null;
  event_type: string;
  actor: string;
  description: string;
  created_at: string;
}

export interface ToastMessage {
  id: string;
  type: 'success' | 'error' | 'info';
  title: string;
  message: string;
}
